projects/ngx-datatable/src/lib/components/body/body-cell.component.spec.ts

```diff
@@ -1,28 +1,146 @@
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { Component, ComponentRef, TemplateRef, viewChild, AfterViewInit } from '@angular/core';
 import { ComponentFixture, TestBed, waitForAsync } from '@angular/core/testing';
 
+import { TableColumnInternal } from '../../types/internal.types';
 import { toInternalColumn } from '../../utils/column-helper';
 import { numericIndexGetter } from '../../utils/column-prop-getters';
 import { DataTableBodyCellComponent } from './body-cell.component';
+import { BodyCellHarness } from './testing/body-cell.harness';
+
+@Component({
+  selector: 'mock-cell-template',
+  imports: [DataTableBodyCellComponent],
+  template: `<ng-template #template let-row="row">Custom Cell Template {{ row.id }} </ng-template>
+    <datatable-body-cell [row]="row" [column]="column" /> `
+})
+class MockCellTemplateComponent implements AfterViewInit {
+  row = { id: 1 };
+  column: TableColumnInternal<any> = toInternalColumn([{ prop: 'id' }])[0];
+  readonly template = viewChild('template', { read: TemplateRef<any> });
+
+  ngAfterViewInit() {
+    this.column = {
+      ...this.column,
+      cellTemplate: this.template()
+    };
+  }
+}
 
 describe('DataTableBodyCellComponent', () => {
   let fixture: ComponentFixture<DataTableBodyCellComponent>;
-  let component: DataTableBodyCellComponent;
+  let component: ComponentRef<DataTableBodyCellComponent>;
+  let harness: BodyCellHarness;
 
-  beforeEach(waitForAsync(() => {
+  beforeEach(waitForAsync(async () => {
     fixture = TestBed.createComponent(DataTableBodyCellComponent);
-    component = fixture.componentInstance;
+    component = fixture.componentRef;
+    component.setInput('row', ['Hello']);
+    component.setInput('column', {});
+    component.setInput('ariaRowCheckboxMessage', 'checkbox message');
+    harness = await TestbedHarnessEnvironment.harnessForFixture(fixture, BodyCellHarness);
   }));
 
-  describe('prop tests', () => {
-    // verify there wasn't a mistake where the falsey 0 value
-    // resulted in a code path for missing column prop
-    it('should get value from zero-index prop', () => {
-      component.row = ['Hello'];
-      const columns = toInternalColumn([{ name: 'First Column', prop: 0 }]);
-      expect(columns[0].$$valueGetter).toBe(numericIndexGetter);
+  it('should get value from zero-index prop', async () => {
+    const columns = toInternalColumn([{ name: 'First Column', prop: 0 }]);
+    expect(columns[0].$$valueGetter).toBe(numericIndexGetter);
+
+    component.setInput('column', columns[0]);
+
+    const cellText = await harness.getBodyCellText();
+    expect(cellText).toEqual('Hello');
+  });
+
+  it('should have min width when column has minWidth', async () => {
+    const columns = toInternalColumn([{ name: 'Min Width Column', prop: 0, minWidth: 500 }]);
+    component.setInput('column', columns[0]);
+
+    const cellElement = await harness.bodyCellWidth();
+    expect(cellElement).toBe(500);
```

Comment:
Please inline all of those calls
```suggestion
    expect(await harness.bodyCellWidth()).toBe(500);
```